p2p/transport/webtransport/transport.go

```diff
@@ -268,3 +275,51 @@ func (t *transport) Close() error {
 	}
 	return nil
 }
+
+// extractSNI returns what the SNI should be for the given maddr. If there is an
+// SNI component in the multiaddr, then it will be returned and
+// foundSniComponent will be true. If there's no SNI component, but there is a
+// DNS-like compoent, then that will be returned for the sni and
+// foundSniComponent will be false (since we didn't find an actual sni component).
+func extractSNI(maddr ma.Multiaddr) (sni string, foundSniComponent bool) {
```

Comment:
Do we need the `bool` here? Would an empty string be sufficient? As far as I know, there's no way to distinguish between empty string and unset in the TLS ClientHello: https://github.com/golang/go/blob/d74bf73be052851e83fb59a40f47d49f4b890ca3/src/crypto/tls/handshake_messages.go#L124-L135.

I think the more idiomatic variable name for the `bool` would be `ok`.

Reply:
Not carefully enough, apparently ;)